### PR TITLE
Hangout thread

### LINKED ISSUE #9784: Custom Hangouts alerts
##### Feature idea summary

Google Hangouts notifications are a great mechanism to notify people, but the current behaviour sends all the information as an attachment that is not visible in the notification of mobile devices.

It would be more useful to have a line of text with the summary for the alert (Host, Context, Level, etc.) at the top of the message and then the attachment, that way mobile devices would get a notification that is readable at a glance.

##### Expected behavior

Ability to customise the content of alarms sent to Hangouts.

### LINKED ISSUE #9785: Threads for Hangouts Alerts
##### Feature idea summary
Instead of each new alert creating a new thread, it would be useful to have pre-created threads in Google Hangouts and configure them in NetData so it appends new messages to them, to create better traceability of an issue.

##### Expected behavior
The configuration file could have, in addition to the URL for the webhook, a field for the thread, that if provided will be used for all the messages of that particular alert. This way the messages will be linked in a single thread which provides readability.

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
Fixes #9784 
Fixes #9785 

This PR brings a new feature for Hangout notification.
##### Component Name
Health
##### Test Plan
1 - Create a room on Hangout.
2 - Create a webhook.
3 - Inside the room, create a specific thread and copy the link
4 - Configure the `healt_alarm_notify.conf` file :

```
SEND_HANGOUTS="YES"
HANGOUTS_WEBHOOK_URI[Alarms]="https://chat.googleapis.com/v1/spaces/AAAAxxxxxxx/messages?key=XXXXX&token=XXXXX"
#HANGOUTS_WEBHOOK_THREAD[Alarms]="spaces/XXXXXXXXX/threads/YYYYYYY"
DEFAULT_RECIPIENT_HANGOUTS="Alarms"
```

for the first test, we will keep the  variable `HANGOUTS_WEBHOOK_THREAD[Alarms]` commented.

4 - Run the following commands:

```
$ cd /usr/libexec/netdata/plugins.d/
$ ./alarm-notfy.sh test
```

the final result will be:

independent

5 - Uncoment the line `HANGOUTS_WEBHOOK_THREAD[Alarms]="spaces/XXXXXXXXX/threads/YYYYYYY"` and run again:

```
$ ./alarm-notfy.sh test
```

The expected result is:

together

### CODE REVIEW

File: health/notifications/alarm-notify.sh.in [hunk 1]
```diff
@@ -1901,6 +1901,9 @@ send_hangouts() {
 		if [ -z "${HANGOUTS_WEBHOOK_URI[$room]}" ] ; then
 			info "Can't send Hangouts notification for: ${host} ${chart}.${name} to room ${room}. HANGOUTS_WEBHOOK_URI[$room] not defined"
 		else
+			if [ -n "${HANGOUTS_WEBHOOK_THREAD[$room]}" ]; then
```
Comment (ilyam8) [inline health/notifications/alarm-notify.sh.in:1904]: do we need to `declare HANGOUTS_WEBHOOK_THREAD`?

like https://github.com/thiagoftsm/netdata/blob/0b979a2fe45e4434b260141fa88c4729556984ce/health/notifications/alarm-notify.sh.in#L375-L376

File: health/notifications/hangouts/README.md [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
+description: "Send alerts to Send notifications to Google Hangouts any time an anomaly or performance issue strikes a node in your infrastructure."
+sidebar_label: "Google Hangouts"
 custom_edit_url: https://github.com/netdata/netdata/edit/master/health/notifications/hangouts/README.md
----
 -->
 
-# Hangouts Chat
+# Send notifications to Google Hangouts
+
+[Google Hangouts](https://hangouts.google.com/) is cross-platform messaging app developed by Google. You can configure
```
Comment (vlvkobal) [inline health/notifications/hangouts/README.md:10]: is `a` cross-platform messaging app

File: health/notifications/hangouts/README.md [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
 
-How to create an incoming webhook: 
-https://developers.google.com/hangouts/chat/how-tos/webhooks
+Set the webhook URIs and room names in `health_alarm_notify.conf`. To edit it on your system, run
+`/etc/netdata/edit-config health_alarm_notify.conf`):
 
-Set the webhook URIs and room names in `health_alarm_notify.conf`. To edit it on your system, run `/etc/netdata/edit-config health_alarm_notify.conf`):
+## Threads (optional)
+
+Instead to receive alarms on different threads, Netdata allows you to concentrate them inside an unique thread when you
```
Comment (vlvkobal) [inline health/notifications/hangouts/README.md:29]: Instead `of receiving`
`a` unique

File: health/notifications/alarm-notify.sh.in (hunk 1 above)
Comment (thiagoftsm) [reply health/notifications/alarm-notify.sh.in:1905]: Thank you @ilyam8 !

File: health/notifications/hangouts/README.md (hunk 3 above)
Comment (thiagoftsm) [reply health/notifications/hangouts/README.md:29]: @joelhans you reviewed yesterday, please, can you confirm these @vlvkobal suggestions?

Comment (vlvkobal) [reply health/notifications/hangouts/README.md:29]: I think Joel knows better than me 😉

**Review by vlvkobal — APPROVED** [review]

**Review by ilyam8 — APPROVED** [review]

**Pull request merged**